sycl/doc/extensions/MaxWorkGroupQueries/max_work_group_query.md

```diff
@@ -0,0 +1,81 @@
+# SYCL_EXT_ONEAPI_MAX_WORK_GROUP_QUERY
+
+## Introduction
+
+This extension adds two new device information descriptors. They provide the ability to query a device for the maximum numbers of work-groups that can be submitted in each dimension as well as globally (across all dimensions).
+
+OpenCL never offered such query - which is probably why it is absent from SYCL. Now that SYCL supports back-ends where the maximum number of work-groups in each dimension can be different, having the ability to query that limit is crucial in writing safe and portable code.
+
+## Feature test macro
+
+As encouraged by the SYCL specification, a feature-test macro, `SYCL_EXT_ONEAPI_MAX_WORK_GROUP_QUERY`, is provided to determine whether this extension is implemented.
+
+## New device descriptors
+
+| Device descriptors                                     | Return type | Description                                                                                                                                                                                                          |
+| ------------------------------------------------------ | ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| info::device::ext_oneapi_max_number_work_groups        |  id<3>      | Returns the maximum number of work-groups that can be submitted in each dimension of the `globalSize` of a `nd_range`. The minimum value is `(1, 1, 1)` if the device is different than `info::device_type::custom`. |
```

Comment:
The semantics of this query are unclear if the `nd_range` is 1- or 2-dimensional.  For example, do you ignore the first element or the last element for a 2-dimensional `nd_range`?

One way to solve this is to use three queries:

* `ext_oneapi_max_number_work_groups_1d`
* `ext_oneapi_max_number_work_groups_2d`
* `ext_oneapi_max_number_work_groups_3d`

This is also advantageous because you say below that that the future direction is to make `ext_oneapi_max_number_work_groups` a templated query.  The API you have now will make this difficult without breaking backward compatibility.  Applications coded to the proposed API will use `ext_oneapi_max_number_work_groups` as an untemplated query, but it might be difficult to support these applications in the future if `ext_oneapi_max_number_work_groups` becomes a templated query.  However, if the first version of this API uses `ext_oneapi_max_number_work_groups_1d`, etc., then you can continue to support these queries even after introducing `ext_oneapi_max_number_work_groups` as a templated query.

Reply:
I acknowledge it's unclear, I wasn't sure how to express that. For a `nd_range<2>` we would discard the first element, and for a `nd_range<1>` discard the first two. 
I added the three queries so the templated version will be straightforward. To avoid code deduplication, the PI backends just implements `PI_EXT_ONEAPI_DEVICE_INFO_MAX_NUMBER_WORK_GROUPS_3D` and then the values are rearranged in device_info.hpp